File: frontend/src/types/decision.ts
/** 投资决策相关类型 */

export interface DecisionFactorDetail {
  score: number
  weight: number
  label: string
  details: Record<string, any>
}

export interface DecisionFactors {
  technical_score: number
  sentiment_score: number
  risk_score: number
  momentum_score: number
  fundamental_score: number
  composite_score: number
  technical: DecisionFactorDetail
  sentiment: DecisionFactorDetail
  risk: DecisionFactorDetail
  momentum: DecisionFactorDetail
  fundamental?: DecisionFactorDetail
  regime?: string
  weights?: Record<string, number>
  weekly_technical?: DecisionFactorDetail
}

export interface InvestmentDecision {
  id: number
  user_id: number
  symbol: string
  recommendation: 'strong_buy' | 'buy' | 'hold' | 'sell' | 'strong_sell'
  confidence: number
  target_price: number | null
  stop_loss: number | null
  factors: DecisionFactors | null
  reasoning: string | null
  status: 'active' | 'executed' | 'dismissed' | 'expired'
  valid_until: string | null
  created_at: string
  updated_at: string
}

export interface DecisionSummary {
  total_active: number
  strong_buy_count: number
  buy_count: number
  hold_count: number
  sell_count: number
  strong_sell_count: number
  avg_confidence: number
  top_picks: InvestmentDecision[]
  recent_decisions: InvestmentDecision[]
}

export interface DecisionOutcome {
  id: number
  decision_id: number
  symbol: string
  recommendation: string
  confidence: number
  entry_price: number | null
  actual_high_24h: number | null
  actual_low_24h: number | null
  actual_close_24h: number | null
  hit_target: boolean
  hit_stop: boolean
  pnl_pct: number | null
  outcome: 'win' | 'loss' | 'breakeven' | null
  checked_at: string | null
}

export interface OutcomeSummary {
  total: number
  wins: number
  losses: number
  breakeven_count: number
  win_rate: number
  avg_pnl_pct: number
  strong_buy_accuracy: number
  buy_accuracy: number
  hold_accuracy: number
  sell_accuracy: number
  strong_sell_accuracy: number
  recent_outcomes: DecisionOutcome[]
}

export const recommendationConfig: Record<string, { label: string; type: 'success' | 'warning' | 'danger' | 'info' | '' }> = {
  strong_buy: { label: '强烈买入', type: 'success' },
  buy: { label: '买入', type: 'success' },
  hold: { label: '持有', type: 'info' },
  sell: { label: '卖出', type: 'danger' },
  strong_sell: { label: '强烈卖出', type: 'danger' },
}

export const outcomeConfig: Record<string, { label: string; type: 'success' | 'warning' | 'danger' | 'info' }> = {
  win: { label: '正确', type: 'success' },
  loss: { label: '错误', type: 'danger' },
  breakeven: { label: '平盘', type: 'info' },
}

export const regimeLabels: Record<string, string> = {
  trending_up: '上涨趋势',
  trending_down: '下跌趋势',
  ranging: '震荡整理',
  volatile: '高波动',
}

File: backend/app/schemas/decision.py
"""
投资决策建议 Pydantic Schema
"""
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class DecisionGenerateRequest(BaseModel):
    """生成决策请求"""
    symbols: List[str] = Field(..., min_length=1, description="股票代码列表")


class DecisionFactorDetail(BaseModel):
    """因子详情"""
    score: float = Field(description="因子得分 0-100")
    weight: float = Field(description="因子权重")
    label: str = Field(description="因子名称")
    details: Dict[str, Any] = Field(default_factory=dict, description="因子详细数据")


class DecisionFactors(BaseModel):
    """多因子分析结果"""
    technical_score: float
    sentiment_score: float
    risk_score: float
    momentum_score: float
    fundamental_score: float = 0.0
    composite_score: float
    technical: DecisionFactorDetail
    sentiment: DecisionFactorDetail
    risk: DecisionFactorDetail
    momentum: DecisionFactorDetail
    fundamental: Optional[DecisionFactorDetail] = None
    regime: Optional[str] = None
    weights: Optional[Dict[str, float]] = None
    weekly_technical: Optional[DecisionFactorDetail] = None


class DecisionResponse(BaseModel):
    """决策响应"""
    id: int
    user_id: int
    symbol: str
    recommendation: str
    confidence: int
    target_price: Optional[float] = None
    stop_loss: Optional[float] = None
    factors: Optional[Dict[str, Any]] = None
    reasoning: Optional[str] = None
    status: str
    valid_until: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DecisionSummaryResponse(BaseModel):
    """决策汇总统计"""
    total_active: int = 0
    strong_buy_count: int = 0
    buy_count: int = 0
    hold_count: int = 0
    sell_count: int = 0
    strong_sell_count: int = 0
    avg_confidence: float = 0.0
    top_picks: List[DecisionResponse] = Field(default_factory=list)
    recent_decisions: List[DecisionResponse] = Field(default_factory=list)


# ---------------------------------------------------------------------------
# Outcome tracking schemas (P3)
# ---------------------------------------------------------------------------


class DecisionOutcomeResponse(BaseModel):
    """决策结果响应"""
    id: int
    decision_id: int
    symbol: str
    recommendation: str
    confidence: int
    entry_price: Optional[float] = None
    actual_high_24h: Optional[float] = None
    actual_low_24h: Optional[float] = None
    actual_close_24h: Optional[float] = None
    hit_target: bool = False
    hit_stop: bool = False
    pnl_pct: Optional[float] = None
    outcome: Optional[str] = None
    checked_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class OutcomeSummaryResponse(BaseModel):
    """准确率汇总"""
    total: int = 0
    wins: int = 0
    losses: int = 0
    breakeven_count: int = 0
    win_rate: float = 0.0
    avg_pnl_pct: float = 0.0
    strong_buy_accuracy: float = 0.0
    buy_accuracy: float = 0.0
    hold_accuracy: float = 0.0
    sell_accuracy: float = 0.0
    strong_sell_accuracy: float = 0.0
    recent_outcomes: List[DecisionOutcomeResponse] = Field(default_factory=list)
